### ISSUE TITLE
Issues with sparse unique indexes using YAML mapping 

### ISSUE THREAD

**Issue by AliHichem** [doctrine/mongodb-odm#1033]
Hi ,
First of all thanks for this awesome ODM !
I'm having a trouble when creating unique indexes using yml mapping files, for instance if I want a unique index on "sid" with sparse option active I only get the unique index without sparse. My "fields" section look like this
```
fields:
      id:
        type : id
        id:  true
        strategy: INCREMENT
      sid:
        type: string
        unique: true
        sparse: true
```
After updating the schema and if I check the indexes the "sparse" option is never added :
```
> db.getCollection('Messaging\\Message').getIndexes();
[
	{
		"v" : 1,
		"key" : {
			"_id" : 1
		},
		"name" : "_id_",
		"ns" : "mydb.Messaging\\Message"
	},
	{
		"v" : 1,
		"unique" : true,
		"key" : {
			"sid" : 1
		},
		"name" : "sid_1",
		"ns" : "mydb.Messaging\\Message",
		"w" : 1
	}
]
```
This is not a mongodb limitation because I can create unique sparse indexes with command line : 
```
> db.getCollection('Messaging\\Message').ensureIndex({'sid':1},{sparse:1,unique:1});
> db.getCollection('Messaging\\Message').getIndexes();
[
	{
		"v" : 1,
		"key" : {
			"_id" : 1
		},
		"name" : "_id_",
		"ns" : "mydb.Messaging\\Message"
	},
	{
		"v" : 1,
		"key" : {
			"sid" : 1
		},
		"name" : "sid_1",
		"ns" : "mydb.Messaging\\Message",
		"sparse" : true,
		"w" : 1
	}
]
```
Also adding the keyword "index:true" to the "sid" field will just create a nonsense index with neither "unique" nor "sparse" options active...

I'm not sure about the other mapping types, but for yml this is not working.
Thanks for the help !

**Comment by malarzm** [comment]
Yaml driver seems to be the only one affected by looking at the code, I'll try to prepare fix for it tomorrow

**Comment by malarzm** [comment]
@AliHichem but anyway your second console doesn't have ``unique: true`` in index information?

**Comment by AliHichem** [comment]
My bad I did a copy/paste by mistake on a different screen, I just created the index manually again and here's what I have :
```
> db.getCollection('Messaging\\Message').ensureIndex({'sid':1},{sparse:1,unique:1});
{
	"createdCollectionAutomatically" : true,
	"numIndexesBefore" : 1,
	"numIndexesAfter" : 2,
	"ok" : 1
}

> db.getCollection('Messaging\\Message').getIndexes();
[
	{
		"v" : 1,
		"key" : {
			"_id" : 1
		},
		"name" : "_id_",
		"ns" : "mydb.Messaging\\Message"
	},
	{
		"v" : 1,
		"unique" : true,
		"key" : {
			"sid" : 1
		},
		"name" : "sid_1",
		"ns" : "mydb.Messaging\\Message",
		"sparse" : 1
	}
]
```

So once again, if I create the indexes manually I get the unique index with sparse option working perfectly. One thing I'm not sure about is why I don't see the index unique on the id ...But when I wheck the uniqueness on the id it's working , anyway I don't care if it is working :). The only issue I have is with the sparse option .

Thanks for your fast reply !

**Comment by jmikola** [comment]
Fixed in 30f42aa01ff4aa289c1090d6cc7fd79d1ac8afe1.

@AliHichem: In the future, feel free to leave this open until the fix is committed.
